### DISCUSSION TITLE
Deploy features into standalone application

### DISCUSSION THREAD

**Discussion by stephan-strate** [deployphp/deployer#2421] [Help needed]
My goal is to set up deployer to automatically deploy our [Grav](https://getgrav.org/) websites.
After studying the documentation and some tutorials, there are still some questions in my head. Especially how to deploy features into a standalone application. What does this mean? We have got a cms like Typo3 or in this case Grav, which is already "installed" (lives in the public dir) on the webserver. The folder structure looks something like this:
```
/assets
/backup
/bin
/cache
/images
/logs
/system
/tmp
/vendor
/user
index.php
+ some additional files like .htaccess
```

Our git repositories have the same structure, but without the application itself of course. That means our git repositories look something like this:
```
/user
.htaccess
+ some additional files
```

How would this be deployed using deployer? I can not use tasks like `update_code`, because this would need to be executed in an empty directory. You might not support Grav, but Typo3 raises the same question for me.

**Comment by antonmedv** [comment]
Just depends on what you would like to do. For example clone app and then rsync your file. Common recipe is using symlinks for public dir.
